src/openms/include/OpenMS/DATASTRUCTURES/ParamValue.h

```diff
@@ -0,0 +1,396 @@
+    ParamValue& operator=(const float);
+    /// specific assignment for short int values (note: the implementation uses SignedSize)
+    ParamValue& operator=(const short int);
+    /// specific assignment for unsigned short int values (note: the implementation uses SignedSize)
+    ParamValue& operator=(const unsigned short int);
+    /// specific assignment for int values (note: the implementation uses SignedSize)
+    ParamValue& operator=(const int);
+    /// specific assignment for unsigned int values (note: the implementation uses SignedSize)
+    ParamValue& operator=(const unsigned);
+    /// specific assignment for long int values (note: the implementation uses SignedSize)
+    ParamValue& operator=(const long int);
+    /// specific assignment for unsigned long int values (note: the implementation uses SignedSize)
+    ParamValue& operator=(const unsigned long);
+    /// specific assignment for long long int values (note: the implementation uses SignedSize)
+    ParamValue& operator=(const long long);
+    /// specific assignment for unsigned long long int values (note: the implementation uses SignedSize)
+    ParamValue& operator=(const unsigned long long);
+    //@}
+
+    /// returns the type of value stored
+    inline DataType valueType() const
+    {
+      return value_type_;
+    }
+
+    /**
+       @brief Test if the value is empty
+
+       @note A ParamValue containing an empty string ("") does not count as empty!
+    */
+    inline bool isEmpty() const
+    {
+      return value_type_ == EMPTY_VALUE;
+    }
+
+    /// output stream operator
+    friend OPENMS_DLLAPI std::ostream& operator<<(std::ostream&, const ParamValue&);
+
+    /// Equality comparator
+    friend OPENMS_DLLAPI bool operator==(const ParamValue&, const ParamValue&);
+
+    /// Smaller than comparator (for vectors we use the size)
+    friend OPENMS_DLLAPI bool operator<(const ParamValue&, const ParamValue&);
+
+    /// Greater than comparator (for vectors we use the size)
+    friend OPENMS_DLLAPI bool operator>(const ParamValue&, const ParamValue&);
+
+    /// Equality comparator
+    friend OPENMS_DLLAPI bool operator!=(const ParamValue&, const ParamValue&);
+
+protected:
+
+    /// Type of the currently stored value
+    DataType value_type_;
+
+    /// Space to store the data
+    union
+    {
+      ptrdiff_t ssize_;
+      double dou_;
+      std::string* str_;
+      std::vector<std::string>* str_list_;
+      std::vector<int>* int_list_;
+      std::vector<double>* dou_list_;
+    } data_;
+
+private:
+
+    /// Clears the current state of the ParamValue and release every used memory.
+    void clear_() noexcept;
+
+    /// Convert a double to std::string
+    /// with full precision 15 decimal places are given, otherwise 3
+    /// numbers above 10000 or below 0.0001 are given in scientific notation (i.e. 1.0e04)
+    static std::string doubleToString(double value, bool full_precision = true);
```

Comment:
I guess we can't use https://en.cppreference.com/w/cpp/utility/to_chars or https://en.cppreference.com/w/cpp/io/c/fprintf right?

Reply:
I will look into your suggestions if they can be used instead, since i am not really happy with the current solution. I just wanted to make something that works for now.